modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts

```diff
@@ -307,15 +282,15 @@ const removeCaretFormat = (editor: Editor, name: string, vars?: FormatVars, simi
     selection.moveToBookmark(bookmark);
   } else {
     const caretContainer = getParentCaretContainer(editor.getBody(), formatNode);
+    const parentsAfter = Type.isNonNullable(caretContainer) ? dom.getParents(formatNode.parentNode, Fun.always, caretContainer) : [];
     const newCaretContainer = createCaretContainer(false).dom;
 
     insertCaretContainerNode(editor, newCaretContainer, caretContainer ?? formatNode);
 
     const cleanedFormatNode = cleanFormatNode(editor, newCaretContainer, formatNode, name, vars, similar);
-    const caretTextNode = insertFormatNodesIntoCaretContainer(parents.concat(cleanedFormatNode.toArray()), newCaretContainer);
-
+    const caretTextNode = insertFormatNodesIntoCaretContainer(parents.concat(cleanedFormatNode.toArray()).concat(parentsAfter), newCaretContainer);
```

Comment:
```suggestion
    const caretTextNode = insertFormatNodesIntoCaretContainer([
    ...parents,
    ...cleanedFormatNode.toArray(),
    ...parentsAfter ], newCaretContainer);
```
This would be more readable as a spread.